### PR TITLE
Till/2400 message banner

### PR DESCRIPTION
**Pull request by tillprochaska**
* Messages can have a title, a date, and a level (info, warning, error, success).
* Messages can be loaded dynamically from a remote JSON endpoint (configured via `ALEPH_APP_MESSAGES_URL`) or set to a static value via the `ALEPH_APP_BANNER` variable.
* If a JSON endpoint is configured, the app will poll for changes every 15 minutes. This way, even users that keep the Aleph session open for a long time will be shown new messages with a max. 15 minutes delay.
* Messages can be hidden after a fixed amount of time. This can be useful to display a messages for some time once an incident has be resolved.
* Messages can have updates. If a message has updates, the most recent update (and the time it was posted) is displayed.

**A few things to consider:**
* The messages JSON endpoint can be configured via an environment variable. The value of that variable is exposed to the frontend via the app metadata. This isn’t optimal for two reasons:
  1. It creates a request waterfall, i.e. the metadata has to be loaded before being able to load messages. This only happens once on app startup and honestly this also isn’t the worst performance bottleneck, so this should be fine.
  2. If the metadata endpoint is unavailable (e.g. because the API service crashed), the frontend won’t be able to load messages, i.e. we can’t display a message banner in case of a full outage.
  
     The only way to change that would be to include the messages endpoint in the frontend image. That means that everyone who’d like to use this feature would have to build a customized UI Docker image. Additionally, configuring this feature would be different from configuring everything else.
  
     I don’t think it’s worth the effort. As an alternative generating a simple, static HTML status page with the same information would be much more sensible. Users would be able to find status information there even in case of a full outage.

* I’m unsure about naming. "Messages" is pretty generic. On the other hand, this might be used for things other than incidents/system status (e.g. announcing new features or datasets). We already use the terms alerts, notifications for different concepts.

### CODE REVIEW

File: aleph/views/base_api.py [hunk 1]
```diff
@@ -51,6 +51,7 @@ def _metadata_locale(locale):
             "version": __version__,
             "banner": settings.APP_BANNER,
             "ui_uri": settings.APP_UI_URL,
+            "messages_url": settings.APP_MESSAGES_URL,
```
Comment (tillprochaska) [inline aleph/views/base_api.py:54]: What do we prefer? URI or URL?

**Review by Rosencrantz — COMMENTED** [review]
The implementation works and seems to be well developed, it's obvious that you have thoughts about the implementation, there is testing in place which is awesome.

Where I've been struggling with this implementation is in the setting of the different levels. There doesn't seem to be any documentation on the labels I should be setting.

I determined that type:info, type:minor, and type:major all set different levels. But I could not find that documented anywhere.

I also found that when I set the label type:major that the UI used the warning color, which confused me a bit. 

I recommend that we take a look at the documentation, and ensure that it is clear how a user/admin can set the various types.

File: ui/src/app/Router.jsx [hunk 2]
```diff
@@ -34,24 +40,55 @@ import ExportsScreen from 'src/screens/ExportsScreen/ExportsScreen';
 
 import './Router.scss';
 
+const MESSAGES_INTERVAL = 15 * 60 * 1000; // every 15 minutes
```
[the remark this answers is not in the document]
Comment (tillprochaska) [reply ui/src/app/Router.jsx:43]: Refreshing the messages more frequently seemed a little wasteful. The worst case scenario here is that users will see an update 15 minutes after it has been published, which seemed acceptable to me.

**Comment by tillprochaska** [comment]
> I also found that when I set the label type:major that the UI used the warning color, which was confusing.

That’s a bug! Just pushed a commit to fix it.

**Comment by Rosencrantz** [comment]
Two more things. 1. Convert the review from a draft to a real PR. It's confusing otherwise. 2. Are you really looking to merge this directly into main? Should the endpoint actually be the develop branch?

**Comment by tillprochaska** [comment]
> I determined that type:info, type:minor, and type:major all set different levels. But I could not find that documented anywhere. There is also type:announcement and type:scheduled-maintenance which is in our internal documentation on the wiki.

> Are you really looking to merge this directly into main? Should the endpoint actually be the develop branch?

Ah, yes, I forget that every time. So annoying that GitHub has no option to configure the default target branch for PRs.

File: aleph/views/base_api.py (hunk 1 above)
Comment (Rosencrantz) [reply aleph/views/base_api.py:54]: URL is fine, it seems consistent with the settings portion of the line above, which isn't actually consistent with itself, hmmm.

File: ui/src/app/Router.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Rosencrantz) [reply ui/src/app/Router.jsx:43]: Something I noticed when I was playing with the message banner earlier in the week was that updates to the UI were inconsistent. I reduced the refresh to 60 seconds and even accounting for the build process is was really hit and miss as to when the update would show. I also couldn't figure out how long the "fixed" green banner would persist?

**Review by Rosencrantz — APPROVED** [review]

**Pull request merged**